DAOS-17344 build: Add dependencies as submodules (#16162)

jolivier23 opened the pull request:
Primary purpose is to bring in title patch.  But in order to do that easily, there are a few other build patches pulled in.  Most of them are build only but one does update libfuse to fused. Google has been using this exclusively for a few months.

DAOS-15136 client: Switch to using libfused (#14077)
DAOS-17117 build: build dependencies as git submodules (#15909)
DAOS-11866 build: Add --build-deps=fetch option (#14125)
DAOS-623 build: Minor build fix (#15749)
DAOS-623 build: Update developer package for fused (#15746)
DAOS-17344 build: Enhance SCons options (#16157)
DAOS-623 ci: Add LLC to ignore list (#15781)

### Steps for the author:

* [ ] Commit message follows the [guidelines](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Appropriate [Features or Test-tag](https://daosio.atlassian.net/wiki/spaces/DC/pages/10984259629/Test+Tags) pragmas were used.
* [ ] Appropriate [Functional Test Stages](https://daosio.atlassian.net/wiki/spaces/DC/pages/12147556353/CI+Functional+Test+Stages) were run.
* [ ] At least two positive code reviews including at least one code owner from each category referenced in the PR.
* [ ] Testing is complete. If necessary, forced-landing label added and a reason added in a comment.

#### After all prior steps are complete:
* [ ] Gatekeeper requested (daos-gatekeeper added as a reviewer).

jolivier23 committed:
DAOS-17344 build: Add dependencies as submodules (#16162)

Primary purpose is to bring in title patch.  But in order to do that
easily, there are a few other build patches pulled in.  Most of them
are build only but one does update libfuse to fused. Google has been
using this exclusively for a few months.

DAOS-15136 client: Switch to using libfused (#14077)
DAOS-17117 build: build dependencies as git submodules (#15909)
DAOS-11866 build: Add --build-deps=fetch option (#14125)
DAOS-623 build: Minor build fix (#15749)
DAOS-623 build: Update developer package for fused (#15746)
DAOS-17344 build: Enhance SCons options (#16157)

Co-authored-by: Ashley Pittman <ashley.m.pittman@intel.com>
Signed-off-by: Nicholas Murphy <ncmurphy@google.com>
Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
DAOS-623 ci: Add LLC to ignore list (#15781)


Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 commented:
I'd put approved_to_merge label except the extra patches were not considered in the approval process

daltonbohning commented on site_scons/components/__init__.py:
```diff
@@ -1,4 +1,6 @@
 # Copyright 2016-2024 Intel Corporation
+# Copyright 2025 Google LLC
```
Looks like you also need this
https://github.com/daos-stack/daos/commit/8905322fe97761cb936450cef5f74f0afbf55c3a

jolivier23 replied:
Thanks, was looking at the other ignore list and not finding it.

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://jenkins-1.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16422/1/execution/node/243/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://jenkins-1.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16422/1/execution/node/340/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://jenkins-1.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16422/1/execution/node/337/log

jolivier23 committed:
Fix bad merge, missing endif

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

daosbuild3 commented:
Test stage Build RPM on EL 9 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16422/3/execution/node/313/log

daosbuild3 commented:
Test stage Build RPM on Leap 15.5 completed with status UNSTABLE.  https://jenkins-3.daos.hpc.amslabs.hpecorp.net//job/daos-stack/job/daos/view/change-requests/job/PR-16422/3/execution/node/291/log

jolivier23 committed:
Fix date in changelog

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Merge remote-tracking branch 'comm/release/2.6' into jvolivie/backport_deps

jolivier23 committed:
Post merge commit

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Merge remote-tracking branch 'comm/release/2.6' into jvolivie/backport_deps

jolivier23 committed:
Post merge commit

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Merge remote-tracking branch 'comm/release/2.6' into jvolivie/backport_deps

jolivier23 committed:
Merge remote-tracking branch 'comm/release/2.6' into jvolivie/backport_deps

jolivier23 committed:
Merge remote-tracking branch 'comm/release/2.6' into jvolivie/backport_deps

jolivier23 committed:
Merge remote-tracking branch 'comm/release/2.6' into jvolivie/backport_deps

jolivier23 committed:
empty commit

Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
DAOS-623 build: Switch back to git clone

Partial revert of previous commit that contained fixes for having
components as submodules which Google is using.
There is no good way to avoid the copy that doesn't pollute submodules
because we apply patches so best to revert back for now.

Change-Id: I27168a621a3dbb091dd7d4548f52812ef03bd7bf
Signed-off-by: Jeff Olivier <jeffolivier@google.com>

jolivier23 committed:
Merge remote-tracking branch 'comm/release/2.6' into jvolivie/backport_deps

Change-Id: Ib2144dfd9bafd9176ad311127da422e4c885e076

daltonbohning reviewed (commented):
@jolivier23 I suggest closing this now as the release/2.6 branch will likely be frozen soon
